flink-table-store-connector/src/main/java/org/apache/flink/table/store/connector/TableStore.java

```diff
@@ -171,6 +183,35 @@ private FileStore buildFileStore() {
                 tableIdentifier, options, user, partitionType, keyType, valueType, mergeFunction);
     }
 
+    private void adjustIndexAndValidate() {
+        if (primaryKeys.length > 0 && partitions.length > 0) {
+            Set<Integer> pkSet = Arrays.stream(primaryKeys).boxed().collect(Collectors.toSet());
```

Comment:
It is better to use List, because primary key and partition both have the order.

Reply:
> It is better to use List, because primary key and partition both have the order.

I think only after pk and partition pass through the sanity check, the order makes a difference. 
When checking the super/subset relationship, the order doesn't matter.